build/dyn_shim_detection.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+import subprocess
+import xml.etree.ElementTree as ET
+
+_log = logging.getLogger("dyn-shim-detection")
+show_version_info = project.getProperty("dyn.shim.trace")
+_log.setLevel(logging.DEBUG if show_version_info else logging.INFO)
+# Same as shimplify.py
+ch = logging.StreamHandler()
+ch.setFormatter(logging.Formatter('%(name)s - %(levelname)s - %(message)s'))
+_log.addHandler(ch)
+
+multi_module_project_dir = project.getProperty("maven.multiModuleProjectDirectory")
+expression = project.getProperty("expression")
+
+def __get_buildver(pom, expression):
+    pom = ET.parse(pom)
+    ns = {"pom": "http://maven.apache.org/POM/4.0.0"}
+    releases = []
+    for profile in pom.findall("//pom:profile/pom:id", ns):
+       if profile.text.startswith("release"):
+            releases.append(profile.text[7:])
```

Comment:
Nit Not a fan of the fragile magic number that makes this harder to read and fragile.  Recommend something like this to make it more robust:
```suggestion
    release_prefix = "release"
    for profile in pom.findall("//pom:profile/pom:id", ns):
       if profile.text.startswith(release_prefix):
            releases.append(profile.text[len(release_prefix):])
```